physical-based-animations.go

```diff
@@ -90,53 +91,108 @@ func run() {
 	prevDt := 0.002
 
 	last := time.Now()
+
+	gui := GUI{
+		win: win,
+	}
+
+	gui.CreateBatch("assets/sprites/spritesheet.png")
+
+	playButton := Button{
+		position:     pixel.V(10, 10),
+		croppingArea: pixel.R(0, 160, 80, 240),
+		bounds:       pixel.R(0, 0, 80, 80),
+		onClick:      HandlePlayClick,
+	}
+
+	gui.NewButton(playButton)
+
+	pauseButton := Button{
+		position:     pixel.V(10, 100),
+		croppingArea: pixel.R(0, 80, 80, 160),
+		bounds:       pixel.R(0, 0, 80, 80),
+		onClick:      HandlePauseClick,
+	}
+
+	gui.NewButton(pauseButton)
+
+	stopButton := Button{
+		position:     pixel.V(10, 190),
+		croppingArea: pixel.R(0, 0, 80, 80),
+		bounds:       pixel.R(0, 0, 80, 80),
+		onClick:      HandleStopClick,
+	}
+
+	gui.NewButton(stopButton)
+
+	cam := pixel.IM.Scaled(camPos, 1.0).Moved(win.Bounds().Center().Sub(camPos))
+
+	win.SetMatrix(cam)
+
+	gui.SetMatrix(cam)
+	gui.BindState(&state)
+	gui.MainLoop()
+	gui.Draw()
+
 	for !win.Closed() {
-		dt := time.Since(last).Seconds()
-		last = time.Now()
-		timeElapsed += dt
 
-		cam := pixel.IM.Scaled(camPos, 1.0).Moved(win.Bounds().Center().Sub(camPos))
-		win.SetMatrix(cam)
-		win.Clear(colornames.Whitesmoke)
+		win.Update()
 
 		batch.Clear()
```

Comment:
Does batch need to be cleared here? Wouldn't it be enough to clear it before adding particles to it (i.e. before `updateParticles` is called)? Or rather it does not need to be cleared when the game is in state `paused`. And same for `win.Update()`; it needs to be called only if the program is running or once after being stopped to draw cleared canvas.

Reply:
I moved `batch.Clear()` before  updateParticles function but when I moved `win.Update()` function the program freezes. We should found out if it is mac related issue. Can you try to move `win.Update()` function as you suggest and let me know if it freezes?

Reply:
It freezes for me too. I thought the `win.Update()` function just swaps the buffers, but it does something more. Forget about moving it.